language/move-prover/src/bytecode_translator.rs

```diff
@@ -759,6 +805,115 @@ impl<'env> ModuleTranslator<'env> {
                 args
             )
         }
+
+        self.writer.unindent();
+        emitln!(self.writer, "}");
+        emitln!(self.writer);
+    }
+
+    /// Spec check body
+    fn generate_function_spec_check_body(
```

Comment:
This is a bunch of new code with some c&p from existing code. Why do we need this extra handling?

The idea would have been that rewritten functions are entirely dealt with in the transformation pipeline, and no or only marginal changes would be needed in the boogie generator.